### PR TITLE
feat: rtk limit order queries

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Adds the following RTK queries for CoW Swap limit orders (https://docs.cow.fi/cow-protocol/reference/apis/orderbook).

- `getLimitOrder`
- `cancelLimitOrders`
- `getOrderStatus`
- `getTrades`
- `getOrders`

Also improves CoW typing across the board by using enums where appropriate.

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/6199

## Risk

Low - none of these queries are currently consumed anywhere 

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

None.

## Testing

N/A

### Engineering

Ensure the implementation looks sane. The _actual_ testing of these endpoints will come in the next PR when we begin implementing them.

### Operations

- [x] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

Not required.

## Screenshots (if applicable)

N/A

### CODE REVIEW

File: packages/swapper/src/swappers/CowSwapper/utils/constants.ts [hunk 1]
```diff
@@ -9,10 +9,6 @@ export const DEFAULT_ADDRESS = zeroAddress
 export const COW_SWAP_VAULT_RELAYER_ADDRESS = '0xc92e8bdf79f0507f65a392b0ab4667716bfe0110'
 export const COW_SWAP_SETTLEMENT_ADDRESS = '0x9008D19f58AAbD9eD0D60971565AA8510560ab41'
 
-export const ORDER_KIND_SELL = 'sell'
-export const SIGNING_SCHEME = 'eip712'
-export const ERC20_TOKEN_BALANCE = 'erc20'
-
```
Comment (woodenfurniture) [inline packages/swapper/src/swappers/CowSwapper/utils/constants.ts:15]: ❤️

File: src/state/apis/limit-orders/limitOrderApi.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        const maybeNetwork = getCowswapNetwork(payload.chainId)
+        if (maybeNetwork.isErr()) throw maybeNetwork.unwrapErr()
+        const network = maybeNetwork.unwrap()
+        const maybeResult = await cowService.post<LimitOrder>(
+          `${baseUrl}/${network}/api/v1/orders/`,
+          payload,
+        )
+        if (maybeResult.isErr()) throw maybeResult.unwrapErr()
+        const { data: order } = maybeResult.unwrap()
+        return { data: order }
```
Comment (woodenfurniture) [inline src/state/apis/limit-orders/limitOrderApi.ts:39]: [preferably blocking]: These monads are pointless in a non-monadic context, can we use axios without monadic service here like in thorchain swapper `endpoints.ts`?

image

Comment (woodenfurniture) [reply src/state/apis/limit-orders/limitOrderApi.ts:39]: Ignore the promise.all, referring to rawdogging axios

File: src/state/apis/limit-orders/types.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  invalidated: boolean
+  status: OrderExecutionStatus
+  fullFeeAmount: string
+  isLiquidityOrder: boolean
+  ethflowData: EthflowData
+  onchainUser: string
+  onchainOrderData: OnchainOrderData
+  executedSurplusFee: string
+  fullAppData: string
+}
```
Comment (woodenfurniture) [inline src/state/apis/limit-orders/types.ts:145]: [blocking]: Many of these are optional and not guaranteed to be present on the response type, the ones with red `*` are non-optional. e.g `receiver` etc.

Would be worth double checking all the other cow typings as well.

image

Comment (woodenfurniture) [reply src/state/apis/limit-orders/types.ts:145]: If theres any chance our typings dont match cow, we could add myZod parsing for max paranoia

File: src/state/apis/limit-orders/limitOrderApi.ts (hunk 2 above)
Comment (0xApotheosis) [reply src/state/apis/limit-orders/limitOrderApi.ts]: Can do, rawdogged in https://github.com/shapeshift/web/pull/7906/commits/007aadb83e2a7e9ae997fb38a97c2be9d34f26b7

File: src/state/apis/limit-orders/types.ts (hunk 3 above)
Comment (0xApotheosis) [reply src/state/apis/limit-orders/types.ts:163]: Fixed types in https://github.com/shapeshift/web/pull/7906/commits/a5e7934912850aa2b9180d0579ad7e8462b07b26, though have not yet implemented myZod, and won't for this PR. We might want to do a validation PR that does this as a follow-up.

**Review by woodenfurniture — APPROVED** [review]
Quick sanity check of cow trades to ensure they are totally still fine, and they are ✅

**Pull request merged**